src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs

```diff
@@ -0,0 +1,133 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.DotNet.DarcLib.Models.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using NUnit.Framework;
+
+
+namespace Microsoft.DotNet.Darc.Tests.VirtualMonoRepo;
+
+public class VmrRecursiveSyncTests : VmrTestsBase
+{
+    private LocalPath _firstInstallerDependencyPath = null!;
+    private LocalPath _secondInstallerDependencyPath = null!;
+    private string _firstInstallerDependencyName = null!;
+    private string _secondInstallerDependencyName = null!;
+
+    [Test]
+    public async Task RecursiveUpdatePreservesDependencyVersionTest()
+    {
+        await InitializeRepoAtLastCommit(Constants.InstallerRepoName, _installerRepoPath);
+
+        var expectedFilesFromRepos = new List<LocalPath>
+        {
+            _vmrPath / VmrInfo.SourcesDir / Constants.InstallerRepoName / Constants.GetRepoFileName(Constants.InstallerRepoName),
+            _vmrPath / VmrInfo.SourcesDir / _firstInstallerDependencyName / Constants.GetRepoFileName(_firstInstallerDependencyName),
+            _vmrPath / VmrInfo.SourcesDir / _secondInstallerDependencyName / Constants.GetRepoFileName(_secondInstallerDependencyName),
+            _vmrPath / VmrInfo.SourcesDir / Constants.DependencyRepoName / Constants.GetRepoFileName(Constants.DependencyRepoName),
+        };
+
+        var expectedFiles = GetExpectedFilesInVmr(
+            _vmrPath,
+            new[] { Constants.InstallerRepoName, _firstInstallerDependencyName, _secondInstallerDependencyName, Constants.DependencyRepoName },
+            expectedFilesFromRepos);
+
+        CheckDirectoryContents(_vmrPath, expectedFiles);
+
+        // create new version of dependency repo
+
+        File.WriteAllText(_dependencyRepoPath / Constants.DependencyRepoFileName, "New version of the file");
+        await GitOperations.CommitAll(_dependencyRepoPath, "change the file in dependency repo");
+
+        // the second repo depends on the new version, first repo depends on the old one
```

Comment:
Would it be possible to have somewhere a comment with the dependency tree? Similarly as to the original issue?

Reply:
yes, good idea